backend/src/announcement/announcement.service.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { Announcement } from '@prisma/client';
+import { NotFoundError } from '@prisma/client/runtime';
+import { PrismaService } from '../global/prisma/prisma.service';
+import { AnnouncementDTO, ViewAccessENUM } from './dto/announcement.dto';
+
+@Injectable()
+export class AnnouncementServices {
+  constructor(private readonly prisma: PrismaService) {}
+
+  public async getAllAnnouncements(): Promise<AnnouncementDTO[]> {
+    try {
+      const result = await this.prisma.announcement.findMany({});
+      return result.map((announcement) =>
+        AnnouncementServices.convertToDTO(announcement)
+      );
+    } catch (error) {
+      if (error instanceof NotFoundError) {
+        throw new NotFoundException();
+      }
+    }
```

Comment:
since this is a `findMany` query, i dont think it will never throw not found (it would just return an empty list). so i think we can get rid of the try/catch in this case

Reply:
I see. Thank you <3